### PR TITLE
fix oauth null param error

### PR DESCRIPTION
**Pull request by lohanidamodar**
- Fixing issues with optional param in createOAuth2Session reported here https://github.com/appwrite/sdk-for-flutter/issues/23 This occured as we removed default values from SDK
- Fixing issues in client when params is empty while using appwrite client externally as reported here https://github.com/appwrite/sdk-for-flutter/issues/24 - this is for those who are using Client for features that are not yet implemented in the SDK and is under development in Appwrite.

### CODE REVIEW

**Comment by eldadfux** [comment]
@lohanidamodar can you share more about the problem and the fix?

File: templates/flutter/lib/client.dart.twig [hunk 1]
```diff
@@ -134,7 +134,9 @@ class Client {
             await this.init();
         }
 
-        params.removeWhere((key,value) => value == null);
+        if(params.isNotEmpty()) {
```
Comment (JulianBissekkou) [inline templates/flutter/lib/client.dart.twig:137]: In dart this is a getter and not a function ☺️ 

```suggestion
        if(params.isNotEmpty) {
```

File: templates/dart/lib/client.dart.twig [hunk 2]
```diff
@@ -68,7 +68,9 @@ class Client {
             };
         }
 
-        params.removeWhere((key,value) => value == null);
+        if(params.isNotEmpty()) {
```
Comment (JulianBissekkou) [inline templates/dart/lib/client.dart.twig:71]:
```suggestion
        if(params.isNotEmpty) {
```

File: templates/flutter/lib/client.dart.twig (hunk 1 above)
Comment (lohanidamodar) [reply templates/flutter/lib/client.dart.twig:137]: Thanks

**Pull request merged**